client/src/components/home/WishlistIcon.js

```diff
@@ -1,32 +1,24 @@
-import { useEffect, useState } from 'react';
+import { useState } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faHeart } from '@fortawesome/free-solid-svg-icons';
 import { useSelector } from "react-redux";
 import { selectUser } from "../../features/auth/auth.selectors";
-import axios from 'axios';
-
-const removeSeriesFromWishlist = async (series, user) => {
-  await axios.post('http://localhost:8080/wishlist', { action: "REMOVE" , email: user.email, seriesId: series._id});
-  console.log("removing " + JSON.stringify(series) + " from wishlist");
-}
-
-const addSeriesFromWishlist = async(series, user) => {
-  await axios.post('http://localhost:8080/wishlist', { action: "ADD", email: user.email, seriesId: series._id });
-  console.log("adding " + JSON.stringify(series) + " to wishlist");
-
-}
+import wishlistService from "../../services/wishlist.service";
 
 export default function WishlistIcon(props) {
   const [fill, setFill] = useState(false);
   const user = useSelector(selectUser);
 
   const clickHandler = () => {
     if(fill){
-      removeSeriesFromWishlist(props.series, user);
+      wishlistService.updateWishlist(props.series, user, "REMOVE").
+      then(setFill(!fill))
```

Comment:
Did it work? Shouldn't the syntax be `.then(() => setFill(...))` ?
Also can simply it to `setFill(false)` and below `setFill(true)`

Reply:
YES